### PR TITLE
docs(vaults): add docs for Azure Key Vault support in Kong Gateway

### PR DESCRIPTION
**Pull request by jschmid1**
### Description

Add docs for upcoming support for  `Azure Key Vaults` in Kong Gateway.

https://github.com/Kong/kong-ee/pull/6716
https://github.com/Kong/lua-resty-azure/pull/10

EDIT: feature is merged in kong-ee. Docs are ready to be merged

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

https://konghq.atlassian.net/browse/KAG-2532

### CODE REVIEW

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+```json
+{
+  "foo": "bar",
+  "snip": "snap"
+}
+```
+
+Note that Azure AD tenant ID (tenant_id), client ID (client_id), and client secret (client_secret) need to be specified. You can configure these values with environment variables before starting {{site.base_gateway}}:
+
+```bash
```
Comment (jschmid1) [inline app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:57]: Add `VAULT_URI` here when using vaults in `process-secret` (no configured entity mode)

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+export KONG_VAULT_AZURE_TENANT_ID=tenant_id
```
Comment (jschmid1) [inline app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:58]: don't use `KONG` as prefixes

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+export KONG_VAULT_AZURE_CLIENT_ID=client_id
+export KONG_VAULT_AZURE_CLIENT_SECRET=client_secret
+```
+
+```bash
+{vault://azure/secret-name/foo}
```
Comment (Hayk-S) [inline app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:64]: Is the `secret-name` required? I think no as kong will get it from the URI variable
`{vault://azure/foo}`

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md (hunk 1 above)
Comment (jschmid1) [reply app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:57]: The correct way is `KONG_VAULT_AZURE_VAULT_URI`

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md (hunk 3 above)
Comment (jschmid1) [reply app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:64]: if you check the above examples, we create a secret called `secret-name` which contains.

``` json
{
  "foo": "bar",
  "snip": "snap"
}
```

if you want to query `foo` you have to use. 

`{vault://azure/secret-name/foo}`

for `snip` you need to do 

`{vault://azure/secret-name/snip}`

Comment (jschmid1) [reply app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:64]: If you have only a single, non-json value as your secret you can do 

`{vault://azure/secret-name}`

File: app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+* Kong Manager
+* {{site.konnect_short_name}}
+{% endif_version %}
+
+
+Configuration options for a Azure Key Vault in {{site.base_gateway}}:
+
+Parameter | Field name | Description
+----------|------------|------------
+`vaults.config.client_id` | **Client ID** | The client ID from your registered Application. Visit your Azure Dashboard and select **App Registrations** to check your client ID.
```
Comment (jschmid1) [inline app/_src/gateway/kong-enterprise/secrets-management/backends/azure-key-vaults.md:160]: vault_uri is missing

**Comment by Guaris** [comment]
I merged the latest changes from the 3.5 branch into this PR. I'm reviewing it today.

**Review by Guaris — APPROVED** [review]

**Pull request merged**